### PR TITLE
update numpy dependency

### PR DESCRIPTION
**Pull request by rreiner-hqs**
Update numpy dependency.  See [this PR](https://github.com/HQSquantumsimulations/qoqo-quest/pull/192)

### CODE REVIEW

File: qoqo-quest/pyproject.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [project]
 name = "qoqo-quest"
 version = "0.14.5"
-dependencies = ["numpy", "qoqo>=1.16,<1.17", "qoqo_calculator_pyo3>=1.2,<1.3"]
+dependencies = ["numpy>=2.0", "scipy>=1.14", "qoqo>=1.16", "qoqo_calculator_pyo3>=1.2"]
```
Comment (nfwvogt) [inline qoqo-quest/pyproject.toml:4]: Do we need the scipy dependency for anything here?

**Review by kbarkhqs — APPROVED** [review]

File: qoqo-quest/pyproject.toml (hunk 1 above)
Comment (kbarkhqs) [reply qoqo-quest/pyproject.toml:4]: I believe the scipy dependency not being pinned is what caused some of the issues?

Comment (rreiner-hqs) [inline qoqo-quest/pyproject.toml:4]:
```suggestion
dependencies = ["numpy>=2.0", "qoqo>=1.16", "qoqo_calculator_pyo3>=1.2"]
```

Comment (rreiner-hqs) [reply qoqo-quest/pyproject.toml]: I removed it

**Pull request merged**